### ISSUE TITLE
Emit errors from both schema and schema extensions

### ISSUE THREAD

**Issue by rbalicki2** [isographlabs/isograph#139]
Right now, if we encounter any errors when parsing the schema, we emit those errors and do not attempt to parse the schema extensions. But these are independent steps, so we should:

- attempt to parse the schema and schema extension
- emit all errors from those previous steps, if any
- if no errors, proceed

Likewise, the `original_outcome` variable is only dependent on the schema, so we should be able to emit errors from that step and the errors from the schema extension step, etc.

Basically, the code is written linearly, but it should be more of a tree.

-----

One way we might impl that is:

```
let mut errors = vec![];

// instead of unwrapping with ?, we instead call write_errors_to, which gives us back a Result<T, ()>
let type_system_document = read_and_parse_graphql_schema(config).write_errors_to(&mut errors);

// then when we use it, we could do something like
use_type_system_document(
  type_system_document.with_errors(&errors)?
);
```

**Comment by rbalicki2** [comment]
Related to #135

**Comment by rbalicki2** [comment]
Closed in favor of the incremental compiler refactor work
